staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/status_strategy_test.go

```diff
@@ -136,3 +141,106 @@ func TestPrepareForUpdate(t *testing.T) {
 		}
 	}
 }
+
+const listTypeResourceSchema = `
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: foos.test
+spec:
+  group: test
+  names:
+    kind: Foo
+    listKind: FooList
+    plural: foos
+    singular: foo
+  scope: Cluster
+  versions:
+  - name: v1
+    schema:
+      openAPIV3Schema:
+        type: object
+        properties:
+          numArray:
+            type: array
+            x-kubernetes-list-type: set
+            x-kubernetes-map-type: atomic
+            items:
+              type: object
+    served: true
+    storage: true
+`
+
+func TestStatusStrategyValidateUpdate(t *testing.T) {
+	crdV1 := &apiextensionsv1.CustomResourceDefinition{}
+	err := yaml.Unmarshal([]byte(listTypeResourceSchema), &crdV1)
+	if err != nil {
+		t.Fatalf("unexpected decoding error: %v", err)
+	}
+	t.Logf("crd v1 details: %v", crdV1)
+	// set defaults for crdV1
+	apiextensionsv1.SetDefaults_CustomResourceDefinition(crdV1)
+	crd := &apiextensions.CustomResourceDefinition{}
+	if err = apiextensionsv1.Convert_v1_CustomResourceDefinition_To_apiextensions_CustomResourceDefinition(crdV1, crd, nil); err != nil {
```

Comment:
```
	// crdV1.Spec.Versions[0].Schema is still nil after this
	if err = apiextensionsv1.Convert_v1_CustomResourceDefinition_To_apiextensions_CustomResourceDefinition(crdV1, crd, nil); err != nil {
```
That's why I have to convert the json schemaProps below.
```
apiextensionsv1.Convert_v1_JSONSchemaProps_To_apiextensions_JSONSchemaProps(crdV1.Spec.Versions[0].Schema.OpenAPIV3Schema, jsonSchema, nil)
```

Reply:
Fixed. The reason is `validationIdentical` is false.

https://github.com/kubernetes/kubernetes/blob/48c41306c3b4e4fa1661126bfcc2bb4a3c062dee/staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1/conversion.go#L137-L155